app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False
+                 ):
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self):
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self):
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Herbivore):
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
+            herbivore.health -= 50
+        if herbivore.health <= 0:
+            if herbivore in Animal.alive:
+                Animal.alive.remove(herbivore)
```

Comment:
The `bite` method correctly checks if the `herbivore` is an instance of `Herbivore` and not hidden before reducing health. However, the logic for removing the `herbivore` from `Animal.alive` could be simplified by directly checking if `herbivore.health <= 0` and removing it in one step. This is more of a suggestion for clarity rather than a critical issue.